src/Microsoft.OData.Client/Materialization/MaterializerNestedEntry.cs

```diff
@@ -20,6 +20,12 @@ internal class MaterializerNestedEntry : IMaterializerState
         /// <summary>The <see cref="IMaterializerState"/> entries.</summary>
         private readonly List<IMaterializerState> nestedItems;
 
+        /// <summary>The entry.</summary>
+        private MaterializerEntry entry;
+
+        /// <summary>The feed.</summary>
+        private ODataResourceSet feed;
+

```

Comment:
why do we need to add these  here? Since the nestedItems is of type `List<IMaterializerState>` and the `MaterializerEntry` and the `MaterializerFeed` that represents the `ODataResourceSet` already derive from `IMaterializerState`?

Reply:
For backwards compatibility to match how the `MaterializerNavigationLink` worked.